_source/_posts/2022-05-16-jhipster-k8s-azure.md

```diff
@@ -0,0 +1,732 @@
+---
+layout: blog_post
+title: "JHipster Microservice Deployed to Azure with Cosmos DB"
+author: andrew-hughes
+by: contractor
+communities: [devops,security,java,javascript]
+description: ""
+tags: []
+tweets:
+- ""
+- ""
+- ""
+image:
+type: conversion
+---
+## JHipster Azure AKS Microservice
+
+In this tutorial, you'll learn how to deploy a JHipster-based, reactive microservice to Azure Kubernetes Service (AKS). You'll use Azure's Cosmos DB as a persistent store for one of the services. For security, you'll use Okta as an OAuth 2.0 and OpenID Connect (OIDC) provider. You'll also securely encrypt all secrets in the project configuration files using Kubernetes secrets and `kubeseal`. This tutorial focuses on deploying an already generated project to Azure AKS. It does not go into great detail about generating the project. To see how the project was generated using JHipster, take a look at [Reactive Java Microservices with Spring Boot and JHipster](/blog/2021/01/20/reactive-java-microservices). 
+
+The project has a few different pieces:
+
+- JHipster Registry: a Eureka server for service discovery and a Spring Cloud Config server for centralized configuration management
+- Gateway: public gateway application built with Vue
+- Store: store API using Azure's Cosmo DB API for MongoDB
+- Blog: blog API using a Neo4J database
+
+{% img blog/jhipster-k8s-azure/microservice-structure.png alt:"Microservice structure" width:"800" %}{: .center-image }
+
+This tutorial has a lot of different technologies in it. I've tried to make it as simple and as explicit as possible, but it's probably helpful to have some basic knowledge of Docker and Kubernetes before you start.
+
+{% include toc.md %}
+
+If you're already familiar with all the tech in this tutorial, you can skip ahead to the prerequisites section. If not, I'm going to explain them a little before we move on.
+
+## JHipster program structure
+
+**[JHipster](https://www.jhipster.tech/)** is a development platform that streamlines the generation, development, and deployment of both monolithic and microservice applications. It supports a dizzying array of frontend (Angular, React, and Vue) and backend (Spring Boot, Micronaut, Quarkus, Node.js, and .NET) technologies. It's designed to be deployed using Docker and Kubernetes, and can easily deploy to all the major cloud platforms, such as AWS, Azure, Heroku, Cloud Foundry, Google Cloud Platform, and OpenShift. 
```

Comment:
I don't think we should bold links, but a lot of our authors seem to want to do it. What do you think @havi?

Reply:
I'm bolding it to make it stand out in the text to break up the block of text. It just happens to be a link. But maybe since it's already a link, it will stand out enough and doesn't need to be bold.

Reply:
As a general rule, I agree that bolding links can be excessive/distracting. But I like the way bolding is used here to highlight the first mention of a key technology. I think, given the mix of technologies, it's helpful for scannability.